diff -urN psi_old/src/config.h psi/src/config.h
--- psi_old/src/config.h	1970-01-01 05:00:00.000000000 +0500
+++ psi/src/config.h	2008-02-02 20:20:42.156587784 +0500
@@ -0,0 +1 @@
+#define PSI_DATADIR "/usr/share/psi"
Файлы psi_old/src/psi и psi/src/psi различаются
diff -urN psi_old/src/src.pri psi/src/src.pri
--- psi_old/src/src.pri	2008-01-27 23:54:57.530482000 +0500
+++ psi/src/src.pri	2008-02-02 19:15:13.821699928 +0500
@@ -41,6 +41,7 @@
 	CONFIG += tc_psifile
 	mac { CONFIG += tc_itunes }
 	windows { CONFIG += tc_winamp }
+	unix { CONFIG += tc_audacious }
 }
 include($$PWD/tools/tunecontroller/tunecontroller.pri)
 
diff -urN psi_old/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.cpp psi/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.cpp
--- psi_old/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.cpp	1970-01-01 05:00:00.000000000 +0500
+++ psi/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.cpp	2008-02-03 00:23:03.673698780 +0500
@@ -0,0 +1,80 @@
+/*
+ * audaciouscontroller.cpp 
+ * Copyright (C) 2008 Rion
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#include "audaciouscontroller.h"
+
+
+/**
+ * \class AUDACIOUSController
+ * \brief A controller for AUDACIOUS.
+ */
+
+
+/**
+ * \brief Constructs the controller.
+ */
+AUDACIOUSController::AUDACIOUSController() : PollingTuneController()
+{
+	g_type_init();
+	GError *error = NULL;
+	dbus_connection = dbus_g_bus_get(DBUS_BUS_SESSION, &error);
+	if (dbus_connection == NULL)
+		qWarning(qPrintable(QString("Error while connecting to dbus: %1").arg(error->message)));
+	dbus_proxy = dbus_g_proxy_new_for_name(dbus_connection, AUDACIOUS_DBUS_SERVICE,
+                                           AUDACIOUS_DBUS_PATH,
+                                           AUDACIOUS_DBUS_INTERFACE);
+}
+
+
+Tune AUDACIOUSController::currentTune()
+{
+	Tune tune;
+	if (audacious_remote_is_running(dbus_proxy) && audacious_remote_is_playing(dbus_proxy)) {
+		int pos =  audacious_remote_get_playlist_pos(dbus_proxy);
+		
+		gchar* c_title;
+		if (c_title) {
+			c_title = audacious_get_tuple_field_data(dbus_proxy, "title", pos);
+			gchar* c_artist = audacious_get_tuple_field_data(dbus_proxy, "artist", pos);
+			gchar* c_album = audacious_get_tuple_field_data(dbus_proxy, "album", pos);
+			gchar* c_track = audacious_get_tuple_field_data(dbus_proxy, "track-number", pos);
+			if (c_artist) {
+				tune.setArtist(QString(c_artist));
+				free(c_artist);
+			}
+			if (c_album) {
+				tune.setAlbum(QString(c_album));
+				free(c_album);
+			}
+			if (c_track) {
+				tune.setTrack(QString(c_track));
+				free(c_track);
+			}
+		} else {
+			c_title = audacious_remote_get_playlist_file(dbus_proxy, pos);
+		}
+		if (c_title) {
+			tune.setName(QString(c_title));
+			free(c_title);
+		}
+		tune.setTime(audacious_remote_get_playlist_time(dbus_proxy,pos) / 1000);
+	}
+	return tune;
+}
diff -urN psi_old/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.h psi/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.h
--- psi_old/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.h	1970-01-01 05:00:00.000000000 +0500
+++ psi/src/tools/tunecontroller/plugins/audacious/audaciouscontroller.h	2008-02-02 19:59:09.249651014 +0500
@@ -0,0 +1,41 @@
+/*
+ * audaciouscontroller.h
+ * Copyright (C) 2006  Remko Troncon
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#ifndef AUDACIOUSCONTROLLER_H
+#define AUDACIOUSCONTROLLER_H
+
+#include <audacious/audctrl.h>
+#include <audacious/dbus.h>
+
+#include "tunecontrollerinterface.h"
+#include "pollingtunecontroller.h"
+
+class AUDACIOUSController : public PollingTuneController
+{
+public:
+	AUDACIOUSController();
+	
+	virtual Tune currentTune();
+private:
+	DBusGProxy *dbus_proxy;
+	DBusGConnection *dbus_connection;
+};
+
+#endif
diff -urN psi_old/src/tools/tunecontroller/plugins/audacious/audaciousplugin.cpp psi/src/tools/tunecontroller/plugins/audacious/audaciousplugin.cpp
--- psi_old/src/tools/tunecontroller/plugins/audacious/audaciousplugin.cpp	1970-01-01 05:00:00.000000000 +0500
+++ psi/src/tools/tunecontroller/plugins/audacious/audaciousplugin.cpp	2008-02-02 19:24:45.728739278 +0500
@@ -0,0 +1,52 @@
+/*
+ * audaciousplugin.cpp
+ * Copyright (C) 2006  Remko Troncon
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#define QT_STATICPLUGIN
+
+#include <QtCore>
+#include <QObject>
+#include <QString>
+
+#include "tunecontrollerplugin.h"
+#include "audaciouscontroller.h"
+
+class AUDACIOUSPlugin : public QObject, public TuneControllerPlugin
+{
+	Q_OBJECT
+	Q_INTERFACES(TuneControllerPlugin)
+
+public:
+	virtual QString name();
+	virtual TuneController* createController();
+};
+
+Q_EXPORT_PLUGIN2(audaciousplugin, AUDACIOUSPlugin);
+
+QString AUDACIOUSPlugin::name()
+{
+	return "AUDACIOUS";
+}
+
+TuneController* AUDACIOUSPlugin::createController() 
+{
+	return new AUDACIOUSController();
+}
+
+#include "audaciousplugin.moc"
diff -urN psi_old/src/tools/tunecontroller/tunecontrollermanager.cpp psi/src/tools/tunecontroller/tunecontrollermanager.cpp
--- psi_old/src/tools/tunecontroller/tunecontrollermanager.cpp	2007-04-09 02:41:48.341248000 +0600
+++ psi/src/tools/tunecontroller/tunecontrollermanager.cpp	2008-02-02 21:46:37.974772638 +0500
@@ -112,6 +112,10 @@
 Q_IMPORT_PLUGIN(xmmsplugin);
 #endif
 
+#ifdef TC_AUDACIOUS
+Q_IMPORT_PLUGIN(audaciousplugin);
+#endif
+
 #ifdef TC_WINAMP
 Q_IMPORT_PLUGIN(winampplugin);
 #endif
diff -urN psi_old/src/tools/tunecontroller/tunecontroller.pri psi/src/tools/tunecontroller/tunecontroller.pri
--- psi_old/src/tools/tunecontroller/tunecontroller.pri	2007-06-26 01:15:04.410244000 +0600
+++ psi/src/tools/tunecontroller/tunecontroller.pri	2008-02-02 21:06:03.321895410 +0500
@@ -33,6 +33,23 @@
 		$$XMMS_PLUGIN_PATH/xmmsplugin.cpp
 }
 
+# Audacious
+tc_audacious {
+	DEFINES += TC_AUDACIOUS
+	AUDACIOUS_PLUGIN_PATH = $$PWD/plugins/audacious
+	INCLUDEPATH += $$AUDACIOUS_PLUGIN_PATH
+
+	HEADERS += \
+		$$AUDACIOUS_PLUGIN_PATH/audaciouscontroller.h 
+		
+	SOURCES += \
+		$$AUDACIOUS_PLUGIN_PATH/audaciouscontroller.cpp \
+		$$AUDACIOUS_PLUGIN_PATH/audaciousplugin.cpp
+		
+	LIBS += `pkg-config --libs audacious`
+	INCLUDEPATH += /usr/include/glib-2.0 /usr/lib/glib-2.0/include /usr/include/gtk-2.0 /usr/lib/gtk-2.0/include /usr/include/atk-1.0 /usr/include/cairo /usr/include/pango-1.0 /usr/include/dbus-1.0
+}
+
 # iTunes
 tc_itunes {
 	mac {
