Index: src/groupchatdlg.cpp
===================================================================
--- src/groupchatdlg.cpp.orig	2007-11-22 20:25:56.000000000 +0200
+++ src/groupchatdlg.cpp	2007-11-22 20:28:56.000000000 +0200
@@ -83,6 +83,8 @@
 #include <windows.h>
 #endif
 
+#include "groupchattopicdlg.h"
+
 //----------------------------------------------------------------------------
 // GCMainDlg
 //----------------------------------------------------------------------------
@@ -786,18 +788,15 @@
 void GCMainDlg::doTopic()
 {
 	bool ok = false;
-	QString str = QInputDialog::getText(
-		tr("Set Groupchat Topic"),
-		tr("Enter a topic:"),
-		QLineEdit::Normal, ui_.le_topic->text(), &ok, this);
-
-	if(ok) {
+	GroupchatTopicDlg *dialog=new GroupchatTopicDlg(this, ui_.le_topic->text());
+	if (dialog->exec()) {
 		Message m(jid());
 		m.setType("groupchat");
-		m.setSubject(str);
+		m.setSubject(dialog->topic());
 		m.setTimeStamp(QDateTime::currentDateTime());
 		aSend(m);
 	}
+	delete dialog;
 }
 
 void GCMainDlg::doClear()
@@ -1149,7 +1148,10 @@
 	if(!m.subject().isEmpty()) {
 		ui_.le_topic->setText(m.subject());
 		ui_.le_topic->setCursorPosition(0);
-		ui_.le_topic->setToolTip(QString("<qt><p>%1</p></qt>").arg(m.subject()));
+		QString subjectTooltip=m.subject();
+		if (options_->getOption("options.muc.multiline-topic-tooltip").toBool())
+			subjectTooltip.replace("|","<br>");
+		ui_.le_topic->setToolTip(QString("<qt><p>%1</p></qt>").arg(subjectTooltip));	
 		if(m.body().isEmpty()) {
 			if (!from.isEmpty())
 				m.setBody(QString("/me ") + tr("has set the topic to: %1").arg(m.subject()));
Index: src/src.pri
===================================================================
--- src/src.pri.orig	2007-11-22 20:25:56.000000000 +0200
+++ src/src.pri	2007-11-22 20:26:26.000000000 +0200
@@ -148,6 +148,7 @@
 	$$PWD/chateditproxy.h \
 	$$PWD/adduserdlg.h \
 	$$PWD/groupchatdlg.h \
+	$$PWD/groupchattopicdlg.h \
 	$$PWD/gcuserview.h \
 	$$PWD/infodlg.h \
 	$$PWD/translationmanager.h \
@@ -256,6 +257,7 @@
 	$$PWD/tipdlg.cpp \
 	$$PWD/adduserdlg.cpp \
 	$$PWD/groupchatdlg.cpp \
+	$$PWD/groupchattopicdlg.cpp \
 	$$PWD/gcuserview.cpp \
 	$$PWD/infodlg.cpp \
 	$$PWD/translationmanager.cpp \
Index: options/default.xml
===================================================================
--- options/default.xml.orig	2007-11-22 20:24:30.000000000 +0200
+++ options/default.xml	2007-11-22 20:26:26.000000000 +0200
@@ -38,6 +38,7 @@
 			<show-status-changes comment="Show status changes in groupchat window" type="bool">false</show-status-changes>
 			<accept-defaults comment="Automatically accept the default room configuration when a new room is created." type="bool">true</accept-defaults>
 			<auto-configure comment="Automatically open the configuration dialog when a new room is created. This option only has effect if accept-defaults is false." type="bool">true</auto-configure>
+			<multiline-topic-tooltip comment="Replace | in topic tooltip with newline." type="bool">true</multiline-topic-tooltip>
 			<context comment="Options regarding the context being sent when joining a room">
 				<maxchars comment="The maximum number of characters of context that should be sent when entering a room. Use a negative number for an unlimited amount." type="int">-1</maxchars>
 				<maxstanzas comment="The maximum number of context items that should be sent when entering a room. Use a negative number for an unlimited amount." type="int">-1</maxstanzas>
Index: src/groupchattopicdlg.cpp
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ src/groupchattopicdlg.cpp	2007-11-22 20:26:26.000000000 +0200
@@ -0,0 +1,40 @@
+#include "groupchattopicdlg.h"
+
+#include <QLabel>
+#include <QTextEdit>
+#include <QPushButton>
+#include <QVBoxLayout>
+#include <QDialogButtonBox>
+
+GroupchatTopicDlg::GroupchatTopicDlg(QWidget* parent, const QString& oldTopic)
+	:QDialog(parent)
+{
+	setWindowTitle(tr("Set Groupchat Topic"));
+	QLabel *lbl=new QLabel(tr("Enter a topic:"));
+	
+	textEdit=new QTextEdit();
+	QString tmpTopic=oldTopic;
+	tmpTopic.replace(" || ","\n\n");
+	tmpTopic.replace(" | ","\n");
+	textEdit->setPlainText(tmpTopic);
+
+	QDialogButtonBox *buttonBox=new QDialogButtonBox(QDialogButtonBox::Ok
+		| QDialogButtonBox::Cancel);
+	connect(buttonBox, SIGNAL(accepted()), this, SLOT(accept()));
+	connect(buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+
+	QVBoxLayout *layout=new QVBoxLayout();
+	layout->addWidget(lbl);
+	layout->addWidget(textEdit);
+	layout->addWidget(buttonBox);
+
+	setLayout(layout);
+}
+
+QString GroupchatTopicDlg::topic() const
+{
+	QString tmpTopic=textEdit->toPlainText();
+	tmpTopic.replace("\n\n"," || ");
+	tmpTopic.replace("\n"," | ");
+	return tmpTopic;
+}
Index: src/groupchattopicdlg.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ src/groupchattopicdlg.h	2007-11-22 20:26:26.000000000 +0200
@@ -0,0 +1,18 @@
+#ifndef GROUPCHATTOPICDLG_H
+#define GROUPCHATTOPICDLH_H
+
+#include <QDialog>
+
+class QTextEdit;
+
+class GroupchatTopicDlg: public QDialog
+{
+	Q_OBJECT
+public:
+	GroupchatTopicDlg(QWidget* parent, const QString& oldTopic);
+	QString topic() const;
+private:
+	QTextEdit *textEdit;
+};
+
+#endif
